### ISSUE TITLE
Using semantic elements or at least aria-labels for titles

### ISSUE THREAD

**Issue by craigralston** [reddit/reddit#1785]
Navigating subreddits with a screen reader is pretty difficult with the lack of proper HTML5 semantic elements. In a perfect world, I imagine each `.thing` container would be either an `article` or `section` instead of a `div`, which would then allow each section to have its' own heading tags separate from the overall page.

I realize that would be a huge update to make so I will also suggest a much quicker compromise. Using `aria-label` to identify things like the post title would be a big step towards the site being more accessible. It is very difficult to navigate through posts as the current post titles are simply `<p class="title">` - Just a minor change to `<p class="title" aria-label="title">` makes navigating a subreddit much easier on a screen reader.
